server.go

```diff
@@ -265,20 +296,7 @@ func (a *App) runServer(ctx context.Context) {
 			continue
 		}
 
-		switch envelope.Tag {
-		case Reply:
-			if err := a.handleFromReply(ctx, envelope.Message); err != nil {
-				log.Err(err).Msg("handle_from_reply")
-			}
-		case Remote:
-			if err := a.handleFromRemote(ctx, envelope.Message); err != nil {
-				log.Err(err).Msg("handle_from_remote")
-			}
-		case Local:
-			if err := a.handleFromLocal(ctx, envelope.Message); err != nil {
-				log.Err(err).Msg("handle_from_local")
-			}
-		}
+		ch <- envelope
```

Comment:
Actually the select with the ctx.Done() should also be done here 
```suggestion
select{
    case <-ctx.Done():
         return
    case ch <- envelope:
}    
```